### PR TITLE
Add auto padding to models

### PR DESCRIPTION
**Pull request by tourniert**
- Add `AutoPaddingModel` to `SegFormer`, `DeepLabV3`,`DeepLabV3Plus`, `SwinUNetR`, `DeepLabV3` and `UNetRPP`
- Change the shape of input shape in the `test_input_shape_validation()` because `SegFormer` accepts inputs of shape (B, C, 64, 65)
- Remove end of `test_input_shape_validation()` because the `UNetRPP` should know the real input shape in the `__init__()`, which was not the case. Futhermore, the call to `forward()` is already cover in `test_autopad_models`.

### CODE REVIEW

File: mfai/pytorch/models/deeplabv3.py [hunk 1]
```diff
@@ -395,25 +397,38 @@ def check_input_shape(self, x: Tensor) -> None:
...
 
         if self.classification_head is not None:
             labels = self.classification_head(features[-1])
-            return masks, labels
+            return masks, self._maybe_unpadding(labels, old_shape=old_shape)
 
         return masks
 
+    def validate_input_shape(self, input_shape: torch.Size) -> tuple[bool, torch.Size]:
+        # H and W should be divisible by encodre.output_stride
```
Comment (LBerth) [inline mfai/pytorch/models/deeplabv3.py:423]:
```suggestion
        # H and W should be divisible by encoder.output_stride
```

**Comment by LBerth** [comment]
Could we pin the monai lib version in this PR ?

File: mfai/pytorch/models/swinunetr.py [hunk 2]
```diff
@@ -139,3 +149,30 @@ def __init__(
...
+        enc1 = self.encoder2(hidden_states_out[0])
+        enc2 = self.encoder3(hidden_states_out[1])
+        enc3 = self.encoder4(hidden_states_out[2])
+        dec4 = self.encoder10(hidden_states_out[4])
+        dec3 = self.decoder5(dec4, hidden_states_out[3])
+        dec2 = self.decoder4(dec3, enc3)
+        dec1 = self.decoder3(dec2, enc2)
+        dec0 = self.decoder2(dec1, enc1)
+        out = self.decoder1(dec0, enc0)
+        logits = self.out(out)
```
[the remark this answers is not in the document]
Comment (tourniert) [reply mfai/pytorch/models/swinunetr.py:167]: Nice comment, I tougth that I have to rewrite the function because the method `_check_input_size` raise a `ValueError` and not a `RuntimeError` as expected in the test. But I add the `ValueError` to the list of excpection the test expect and simplify the forward.
